pytorch_lightning/plugins/ddp_plugin.py

```diff
@@ -62,3 +62,14 @@ def configure_ddp(self, model, device_ids):
             **self._ddp_kwargs,
         )
         return model
+
+    def input_to_device(self, args: Any, model: LightningModule):
```

Comment:
can you give an example of this? i'm not understanding this method.

Reply:
also, if this call is time dependent, can this be made into a hook?

@ananthsub brought up the point that all accelerators should have the standard hooks as well. that should also be allocated to the next 1 or 2 sprints @edenafek.

maybe `on_before_forward`?

Reply:
yeah that's a neat idea, let me make it a hook for now within the plugin with the name `on_before_forward`! This is what I do in the sharded plugin:

https://github.com/PyTorchLightning/pytorch-lightning/pull/4178/files#diff-49267e3a168db4eb60775a18da157240d63c25cab73eb45e732a7d45db7f0fa8R50

Basically this call is skipped for DDP because DDP handles device movement internally, sharded DDP needs device to be moved up front